modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,224 @@
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: {},
+      hasError: {},
+      isLoaded: false,
+      uploadProgress: -1,
+    };
+
+    this.updateFormElement = this.updateFormElement.bind(this);
+    this.validateAndSubmit = this.validateAndSubmit.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+  }
+
+  componentDidMount() {
+    let self = this;
+    fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then( (resp) => resp.json())
+      .then( (data) => self.setState({data: data, isLoaded: true}))
+      .catch( (error) => {
+        self.setState({error: 'An error occurred when loading the form!'});
+        console.error(error);
+      });
+  }
+
+  render() {
+    // Data loading error
+    if (this.state.error !== undefined) {
+      return (
+        <div className='alert alert-danger text-center'>
+          <strong>
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    return (
+      <FormElement
+        name='uploadfile'
+        fileUpload={true}
+        onSubmit={this.validateAndSubmit}>
+        <FileElement
+          name='file'
+          label='File to upload'
+          onUserInput={this.updateFormElement}
+          required={true}
+          value={this.state.formData.file}
+        />
+        <TextboxElement
+          name='version'
+          label='Version'
+          onUserInput={this.updateFormElement}
+          required={false}
+          value={this.state.formData.version}
+        />
+        <ButtonElement label='Upload File'/>
+        <div className='row'>
+          <div className='col-sm-9 col-sm-offset-3'>
+            <ProgressBar value={this.state.uploadProgress}/>
+          </div>
+        </div>
+      </FormElement>
+    );
+  }
+
+  /**
+   * Update a form Element
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding element
+   */
+  updateFormElement(formElement, value) {
+    let formData = this.state.formData;
+
+    if (value === '') {
+      formData[formElement] = null;
+    } else {
+      formData[formElement] = value;
+    }
+
+    this.setState({formData: formData});
+  }
+
+  /**
+   * Validate and submit the upload
+   */
+  validateAndSubmit() {
+    let files = this.state.data.files ? this.state.data.files : [];
+    let formData = this.state.formData;
+
+    let errorMessage = {
+      Filename: undefined,
+      Filesize: undefined,
+    };
+
+    let hasError = {
+      Filename: undefined,
+      Filesize: undefined,
+    };
+
+    if (!formData.file) {
+      errorMessage.Filename = 'You must select a file to upload';
+      hasError.Filename = true;
+      this.setState({errorMessage, hasError});
+      return;
+    }
+
+    // Check that the size of the file is not bigger than the allowed size
```

Comment:
@johnsaigle re-#5843: yes, that PR adds the check for file size for the data release module.